data/pagination.go

```diff
@@ -0,0 +1,177 @@
+}
+
+func getLimitFromURLQuery(ctx context.Context, cfg *config.Config, query url.Values) int {
+	limit, err := strconv.Atoi(query.Get("limit"))
+	if err != nil {
+		log.Event(ctx, fmt.Sprintf("unable to convert search limit to int - set to default %d", cfg.DefaultLimit), log.INFO)
+		query.Set("limit", strconv.Itoa(cfg.DefaultLimit))
+		limit = cfg.DefaultLimit
+	}
+
+	if limit < cfg.DefaultLimit {
+		query.Set("limit", strconv.Itoa(cfg.DefaultLimit))
+		limit = cfg.DefaultLimit
+	}
+
+	if limit > cfg.DefaultMaximumLimit {
+		query.Set("limit", strconv.Itoa(cfg.DefaultMaximumLimit))
+		limit = cfg.DefaultMaximumLimit
+	}
+
+	return limit
+}
+
+func getPageFromURLQuery(ctx context.Context, cfg *config.Config, query url.Values) int {
+	page, err := strconv.Atoi(query.Get("page"))
+	if err != nil {
+		log.Event(ctx, fmt.Sprintf("unable to convert search page to int - set to default %d", cfg.DefaultPage), log.INFO)
+		query.Set("page", strconv.Itoa(cfg.DefaultPage))
+		page = cfg.DefaultPage
+	}
+
+	if page < 1 {
+		log.Event(ctx, fmt.Sprintf("page number is less than default - default to page %d", cfg.DefaultPage), log.INFO)
+		query.Set("page", strconv.Itoa(cfg.DefaultPage))
+		page = cfg.DefaultPage
+	}
+
+	return page
+}
+
+func getOffset(ctx context.Context, cfg *config.Config, page int, limit int) (offset int, err error) {
+
+	offset = (page - 1) * limit
+
+	// when the offset is negative due to negative current page number or limit
+	if offset < 0 {
+		log.Event(ctx, fmt.Sprintf("offset less than 0 - defaulted to offset %d", cfg.DefaultOffset), log.INFO)
+		offset = cfg.DefaultOffset
+	}
+
+	// when the offset is too big due to big current page number and/or limit
+	if (offset - limit) > cfg.DefaultMaximumSearchResults {
+		err = errs.ErrInvalidPage
+		logData := log.Data{"current_page": page, "limit": limit}
+
+		log.Event(ctx, "offset is too big as large page and/or limit given", log.Error(err), log.ERROR, logData)
+
+		return cfg.DefaultOffset, err
+	}
+
+	return offset, nil
+}
+
+// GetTotalPages gets the total pages of the search results
+func GetTotalPages(limit int, count int) int {
+	return (count + limit - 1) / limit
+}
+
+// GetPagesToDisplay gets all the pages available for the search results
+func GetPagesToDisplay(cfg *config.Config, validatedQueryParams SearchURLParams, totalPages int) []model.PageToDisplay {
+	var pagesToDisplay = make([]model.PageToDisplay, 0)
+
+	currentPage := validatedQueryParams.CurrentPage
+
+	startPage := getStartPage(cfg, currentPage, totalPages)
+
+	endPage := getEndPage(startPage, totalPages)
+
+	controllerQuery := createSearchControllerQuery(validatedQueryParams)
+	query := controllerQuery.Get("q")
+
+	for i := startPage; i <= endPage; i++ {
+		pagesToDisplay = append(pagesToDisplay, model.PageToDisplay{
+			PageNumber: i,
+			URL:        getPageURL(query, i, controllerQuery),
+		})
+	}
+
+	return pagesToDisplay
+}
+
+func getStartPage(cfg *config.Config, currentPage int, totalPages int) int {
```

Comment:
Yes. That's exactly it. And your example is correct as well because we are only showing 5 available pages at the bottom. So if we are on page 5, then it would be 3,4,5,6,7